pkg/apis/psmdb/v1/perconaservermongodbrestore_types.go

```diff
@@ -64,13 +68,68 @@ type PerconaServerMongoDBRestoreList struct {
 	Items           []PerconaServerMongoDBRestore `json:"items"`
 }
 
-func (r *PerconaServerMongoDBRestore) CheckFields() error {
+func (r *PerconaServerMongoDBRestore) CheckFields(cluster *PerconaServerMongoDB) error {
 	if len(r.Spec.ClusterName) == 0 {
 		return fmt.Errorf("spec clusterName field is empty")
 	}
-	if len(r.Spec.BackupName) == 0 && (len(r.Spec.StorageName) == 0 || len(r.Spec.Destination) == 0) {
-		return fmt.Errorf("fields backupName or storageName and destination is empty")
+
+	if r.Spec.PITR == nil {
+		if len(r.Spec.BackupName) == 0 && (len(r.Spec.StorageName) == 0 || len(r.Spec.Destination) == 0) {
+			return errors.New("fields backupName or storageName and destination is empty")
+		}
+
+		return nil
+	}
+
+	if !cluster.Spec.Backup.IsEnabledPITR() && r.Spec.BackupName == "" && r.Spec.StorageName == "" {
```

Comment:
1) what if `r.Spec.StorageName` is empty but storage is passed via `r.Spec.Destination` setting? I think it is still a valid case.

2) also both lines 77 and 84 check the same thing but one check is for PITR another for non-PITR. do we need both of them?

Reply:
1. How should it be used? There is no info in spec and AFAIK this is the only usage of `r.Spec.Destination` over the codebase

2. I can combine it, but it would be a little messy:

`if (r.Spec.PITR == nil || !cluster.Spec.Backup.IsEnabledPITR()) && len(r.Spec.BackupName) == 0 && (len(r.Spec.StorageName) == 0 || len(r.Spec.Destination) == 0) {`

`r.Spec.PITR == nil` - is it Point In Time Restore?
`cluster.Spec.Backup.IsEnabledPITR()` - is PITR enabled in a cluster?

Reply:
1. after some research `r.Spec.Destination` is a premature adding. it is not used right now, but let's check it anyway.
2. I don't understand why it is needed to check PITR here at all

Reply:
changed in 5f54f52abce6e64461a9fce410f44d2eb4dd60be as we discussed